.github/workflows/build.yaml

```diff
@@ -0,0 +1,48 @@
+name: Build
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  pull_request:
+env:
+  go_version: 1.21.7  # Should match the value in arcalot/.github/.../go_lint_and_test.yaml
+jobs:
+  lint_and_test:
+    name: lint and test
+    uses: webbnh/.github/.github/workflows/go_lint_and_test.yaml@common_workflows  # FIXME:  should be arcalot/...@v<tag>
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    needs:
+      - lint_and_test
+    env:
+      for_release: ${{ startsWith(github.event.ref, 'refs/tags/') }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.go_version }}
+      - name: Build ${{ env.for_release == 'true' && 'release' || 'snapshot' }}
```

Comment:
Shouldn't `env.for_release` be a boolean already? Or does `env` stringify the `startsWith` result? 😦 The documentation isn't entirely clear, and it's hard to experiment ...

Reply:
> Shouldn't `env.for_release` be a boolean already?

Yes, I too think it should...but hard experimentation suggests that it's not, presumably because something stringifies it.  (I tried referencing it without the `==` operator, and I tried using `==` with a boolean literal, and neither one worked properly; so, I fell back to the string compare.  😞)

Reply:
That's sad.